### PR TITLE
Improve exported types in expo-sqlite module.

### LINKED ISSUE #5264: Property 'transaction' does not exist on type 'WebSQLDatabase'.
## 🐛 Bug Report

Types missing for SQLite since it was separated into its own package (expo-sqlite). App still runs fine.

### Environment

iOS

  Expo CLI 3.0.9 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.6.0 - /usr/local/bin/node
      npm: 6.10.2 - /usr/local/bin/npm
    IDEs:
      Xcode: 10.3/10G8 - /usr/bin/xcodebuild
    npmPackages:
      @types/expo: ^32.0.13 => 32.0.13 
      @types/react-native: ^0.60.3 => 0.60.3 
      expo: ^34.0.3 => 34.0.4 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.1 => 3.11.1 
    npmGlobalPackages:
      expo-cli: 3.0.9

### Steps to Reproduce

Open a database and execute a transaction (following the example here: https://docs.expo.io/versions/latest/sdk/sqlite/)

```
import { SQLite } from "expo-sqlite";
const db = SQLite.openDatabase(DB_NAME);
db.transaction((tx) => {});
```

### Expected Behavior

No type errors.

### Actual Behavior

We get the following type error: Property 'transaction' does not exist on type 'WebSQLDatabase'.

Looking at the source code (https://github.com/expo/expo/blob/master/packages/expo-sqlite/src/SQLite.ts), I see that the WebSQLDatabase interface only has an `exec` property, but there should also be a `transaction` property.

### PR DESCRIPTION
**Pull request by Szymon20000**
# Why

Resolves:https://github.com/expo/expo/issues/5264

# How

Copy several types from @types/websql.

# Test Plan

<details>

<summary> App.tsx </summary>

```typescript

 import * as React from 'react';
import { ScrollView, Text, View, StyleSheet } from 'react-native';
import * as SQLite from 'expo-sqlite';
import { SQLTransaction, SQLTransactionErrorCallback, SQLError } from 'expo-sqlite';

const db: SQLite.WebSQLDatabase = SQLite.openDatabase('db.db');
const { Component } = React;

export default class App extends React.Component<{}> {
  async componentDidMount() {
    console.log(JSON.stringify(db));
    await this.testTransaction();
    await this.checkPragma();
  }

  async printForeignKeys() {
    console.log("ooooo");
    const p = new Promise((resolve, reject) => {
      db.transaction(
        tx => {
          const onSuccess: SQLite.SQLStatementCallback = (tx: SQLite.SQLTransaction, result: SQLite.SQLResultSet): void => {
            console.log(JSON.stringify(result));
          };

          const onError:  SQLite.SQLStatementErrorCallback = (tx: SQLite.SQLTransaction, error: SQLite.SQLError): boolean => {
            console.log(`Error`, { error });
            return false;
          };

          tx.executeSql(`PRAGMA foreign_keys;`, [], onSuccess, onError);

        },
        () => {
          console.log(`TX fail`);
          reject();
        },
        () => {
          console.log(`TX OK.`);
          resolve();
        }
      );

    });
    return p;
  }

  async checkPragma() {
    console.log("pragma test");

    await this.printForeignKeys();

    await db.exec(
      [{ sql: 'PRAGMA foreign_keys = ON;', args: [] }],
      false,
      () => console.log('Foreign keys turned on'),
    );

    await this.printForeignKeys();
  }

  async testTransaction() {
    console.log('Testing transaction: start.');
    const p = new Promise((resolve, reject) => {
      db.transaction(
        tx => {
          const onSuccess: SQLite.SQLStatementCallback = (tx: SQLite.SQLTransaction, result: SQLite.SQLResultSet): void => {
            console.log(JSON.stringify(result));
          };

          const onError:  SQLite.SQLStatementErrorCallback = (tx: SQLite.SQLTransaction, error: SQLite.SQLError): boolean => {
            console.log(`Error`, { error });
            return false;
          };

          tx.executeSql(`DROP TABLE IF EXISTS abc;`, [], onSuccess, onError);
          tx.executeSql(`CREATE TABLE abc (a, b, c);`, [], onSuccess, onError);
          tx.executeSql(
            `INSERT INTO abc (a, b, c) VALUES (1,2,3);`,
            [],
            onSuccess,
            onError
          );
          tx.executeSql(
            `INSERTSSS INTO abc (a, b, c) VALUES (4,5,6);`,
            [],
            onSuccess,
            onError
          );
        },
        () => {
          console.log(`TX fail`);
          reject();
        },
        () => {
          console.log(`TX OK.`);
          resolve();
        }
      );

      console.log('Testing transaction: end.');
    });
    return p;
  }

  render() {
    return (
      <View style={{ padding: 20 }}>
        <Text>{`Check logs`}</Text>
      </View>
    );
  }
}

```

</details>

### CODE REVIEW

**Review by bbarthec — APPROVED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -12277,6 +12277,30 @@ react-native-tab-view@^1.0.0, react-native-tab-view@^1.2.0, react-native-tab-vie
   dependencies:
     prop-types "^15.6.1"
 
+react-native-unimodules@^0.5.3, react-native-unimodules@~0.5.0, react-native-unimodules@~0.5.2:
```
[the remark this answers is not in the document]
Comment (sjchmiela) [reply yarn.lock]: Have you run `git submodule update --init` recently, @Szymon20000? @tsapeta has recently added `react-native-unimodules` submodule which may not be checked out

**Comment by bbarthec** [comment]
I believe `packages/expo-sqlite/src/SQLite.types.ts` should be cleaned up from all `comments` and also `exports` that are unused and does not correspond to any exported method or functionality.

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
+// Type definitions for websql
+// Project: http://www.w3.org/TR/webdatabase/
+// Definitions by: TeamworkGuy2 <https://github.com/TeamworkGuy2>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// W3C spec: http://www.w3.org/TR/webdatabase/#database
+// Spec revision: 2010-11-18
+// NOTE: the W3C websql spec has been deprecated
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: Above lines: to be removed

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+//[Callback = FunctionOnly, NoInterfaceObject]
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]:
```suggestion
```
and all similar in this file also to be removed

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+export interface DatabaseCallback {
+  /*handleEvent*/ (database: Database): void;
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]:
```suggestion
  (database: Database): void;
```
and all similar

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+/** 4.3 Asynchronous database API - The transaction() and readTransaction() methods takes
+ * one to three arguments. When called, these methods must immediately return and then
+ * asynchronously run the transaction steps with the transaction callback being the
+ * first argument, the error callback being the second argument, if any, the success
+ * callback being the third argument, if any, and with no preflight operation or
+ * postflight operation
+ */
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: It's taken directly from https://www.w3.org/TR/webdatabase/ and I would rather have some comment from https://docs.expo.io

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+export interface Database {
+  /*readonly/const*/ version: string;
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]:
```suggestion
  readonly version: string;
```

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+  transaction(
+    callback: SQLTransactionCallback,
+    errorCallback?: SQLTransactionErrorCallback,
+    successCallback?: SQLVoidCallback
+  ): void;
+
+  readTransaction(
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: Do we really export such native functionality? Nothing like this is mentioned in docs 🤔 TO be remove IMO

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+    successCallback?: SQLVoidCallback
+  ): void;
+
+  /** The changeVersion() method allows scripts to atomically verify the version number and change
+   * it at the same time as doing a schema update. When the method is invoked, it must immediately
+   * return, and then asynchronously run the transaction steps with the transaction callback being
+   * the third argument, the error callback being the fourth argument, the success callback being
+   * the fifth argument
+   */
+  changeVersion(
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: Method to be removed - we do not provide such native functionality, does we?
Exported methods includes:
- https://github.com/expo/expo/blob/master/packages/expo-sqlite/ios/EXSQLite/EXSQLite.m#L67
- https://github.com/expo/expo/blob/master/packages/expo-sqlite/ios/EXSQLite/EXSQLite.m#L92
and
- https://github.com/expo/expo/blob/master/packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.java#L44
- https://github.com/expo/expo/blob/master/packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.java#L78

Comment (bbarthec) [reply packages/expo-sqlite/src/SQLite.types.ts]: Provide types that correspond only to things that we actually provide 😉

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+    oldVersion: string,
+    newVersion: string,
+    callback?: SQLTransactionCallback,
+    errorCallback?: SQLTransactionErrorCallback,
+    successCallback?: SQLVoidCallback
+  ): void;
+}
+
+//[Callback = FunctionOnly, NoInterfaceObject]
+export interface SQLVoidCallback {
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]:
```suggestion
export interface SQLVoidCallback {
  (): void;
}
```
Please, get rid of these meaningless comments in each `***Callback` interface

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+}
+
+//[Callback = FunctionOnly, NoInterfaceObject]
+export interface SQLStatementErrorCallback {
+  /*handleEvent*/ (transaction: SQLTransaction, error: SQLError): boolean;
+}
+
+/** 4.4 Synchronous database API
+ */
+export interface DatabaseSync {
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: We do not use this

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+  static SYNTAX_ERR: number; // = 5;
+  static CONSTRAINT_ERR: number; // = 6;
+  static TIMEOUT_ERR: number; // = 7;
+
+  code: number;
+  message: string;
+}
+
+// synchronous database API error
+export declare class SQLException {
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: It's duplicate of above `SQLError` - get rid of it, eventually preserving comments

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+   * key column duplicated the value of an existing row. */
+  static CONSTRAINT_ERR: number; // = 6;
+  /** Code 7 - A lock for the transaction could not be obtained in a reasonable time. */
+  static TIMEOUT_ERR: number; // = 7;
+
+  code: number;
+  message: string;
+}
+
+export interface WebSQLDatabase extends Database {
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts]: From what I checked, only types from here to the end of file are used, maybe we can shrink all these above to these ones? 😉 
```suggestion
export interface WebSQLDatabase {
  // provide these inherited methods that are actually provided here without interface inheritance
}
```

**Review by bbarthec — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**